internal/commands/app/create_inputs.go

```diff
@@ -83,34 +81,39 @@ func (i *createInputs) resolveDirectory(wd string) (string, error) {
 	if !fi.Mode().IsDir() {
 		return fullPath, nil
 	}
-	_, appOK, appErr := local.FindApp(fullPath)
-	if appErr != nil {
-		return "", appErr
+	_, appOK, err := local.FindApp(fullPath)
+	if err != nil {
+		return "", err
 	}
 	if appOK {
-		return "", fmt.Errorf("%s is inside or is a Realm app directory", dir)
+		return "", errProjectExists{details: fmt.Sprintf("%s is inside or is a Realm app directory", fullPath)}
 	}
 	return fullPath, nil
 }
 
-// TODO(REALMC-8135): Implement data-source flag for app create command
-// func (i *createInputs) resolveDataSource(client realm.Client, groupID, appID string) (string, error) {
-// 	clusters, err := client.ListClusters(groupID, appID)
-// 	if err != nil {
-// 		return "", err
-// 	}
-// 	var dsCluster string
-// 	for _, cluster := range clusters {
-// 		if (i.DataSource == "" && cluster.State == "IDLE") || i.DataSource == cluster.Name {
-// 			dsCluster = cluster.Name
-// 			break
-// 		}
-// 	}
-// 	if dsCluster == "" {
-// 		if i.DataSource != "" {
-// 			return "", fmt.Errorf("Unable to find the %s cluster", i.DataSource)
-// 		}
-// 		return "", fmt.Errorf("Unable to find any available cluster for Group ID %s", groupID)
-// 	}
-// 	return dsCluster, nil
-// }
+func (i *createInputs) resolveDataSource(client realm.Client, groupID, appID string) (map[string]interface{}, error) {
```

Comment:
instead of returning `map[string]interface{}` here, let's just a struct for this (so we can avoid the `["name"]` accessor and use `.Name` instead)

just keep it privately declared here for now (`type dataSource struct { ... }`).  its tempting to say put this in `local` and we can use it as part of the `ServiceStructure`, but I still want to try and keep that `local` structure stuff as generic as possible